aapcs64/aapcs64.rst

```diff
@@ -898,10 +898,44 @@ registers, or if it is a function that returns results in such registers,
 it must ensure that p4-p15 are preserved across the call. In other cases
 it need not preserve any scalable predicate register contents.
 
-Processes, Memory and the Stack
--------------------------------
+Threads and processes
+---------------------
+
+The AAPCS64 applies to a single _`thread` of execution.  Each thread is in
+turn part of a _`process`.  A process might contain one thread or several
+threads.
+
+The exact definition of “thread” and “process” depends on the platform.
+If the platform is a traditional multi-threaded operating system,
+“thread” and “process” generally have their usual meaning for that
+operating system. If the platform supports multiple processes but has
+no separate concept of threads, “thread” is synonymous with “process”.
+If a platform has no concurrency or preemption then it will generally
+have a single “thread” and “process” that executes all instructions.
+
+Each thread has its own register state, defined by the contents of the
+underlying machine registers. A process has a program state defined by
+its threads' register states and by the contents of the memory that the
+process can access. The memory that a process can access, without causing
+a run-time fault, may vary during the execution of its threads.
+
+Memory and the Stack
+--------------------
+
+Memory addresses
+^^^^^^^^^^^^^^^^
+
+The address space may consist of one or more disjoint regions. No region
+may span address zero (although one region may start at zero).
```

Comment:
```suggestion
might span address zero (although one region may start at zero).
```

Reply:
No, we use may consistently with the specific meaning of "permitted, but not required".  This is standard practice in standards documents.

Reply:
could we use 'can' instead of 'may'. May is ambiguous for not-native speakers.

Can start at zero? I

Reply:
"Can" expresses _capability_. "May" expresses _permission_. In an authoritative standard, the two are not the same – it's completely possible for there to be a thing you _can_ do, which is not also a thing that you _may_ do!

Reply:
See my most recent comment.  'May' is the term used in equivalent standards documents for exactly the situation we have here.  In the context of standards it has a well defined and well understood meaning.
These documents are not end-user documentation. They are expected to be read by experienced developers who already understand such terms.  Trying to find alternative words is just going to cause confusion for the target audience.

Reply:
Ok - I'll leave you to make the decision!